backtracking/subset_sum.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file
+ * @brief Program to Count number of subsets
+ *
+ * @details Subset problem (https://en.wikipedia.org/wiki/Subset_sum_problem)
+ * @author [Swastika Gupta](https://github.com/swastyy)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for io operations
+#include <vector>    /// for std::vector
+
+/**
+ * @namespace backtracking
+ * @brief subset sum algorithm
+ */
+namespace backtracking {
+/**
+ * @namespace Subsets
+ * @brief Functions for counting subsets in a given array with a given sum
+ * Time Complexity: O(sum*n), where the sum is the ‘target sum’ and ‘n’ is the
+ * size of the array.
+ */
+namespace Subsets {
+/**
+ * @brief The main function implements count of subsets
+ * @param sum is the required sum of any subset
+ * @param in_arr is the input array
+ * @returns count of the number of subsets with required sum
+ */
+
+std::uint64_t subset_sum(int sum, const std::vector<int> &in_arr) {
+    int nelement = in_arr.size();  // number of subset element
```

Comment:
The comment is wrong and unnecessary
```suggestion
    int nelement = in_arr.size();
```